File: src/programmers/Lv2/올바른괄호/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/12909
function solution(s) {
  let splitS = s.split("");
  let splitSLen = splitS.length - 1;
  let before = [];
  for (let i = 0; i <= splitSLen; i++) {
    if (splitS[i] === "(") {
      before.push(splitS[i]);
    } else {
      if (before[before.length - 1] === "(" && before.length > 0) {
        before.pop();
      } else {
        return false;
      }
    }
  }
  return before.length > 0 ? false : true;
}

File: src/programmers/Lv2/올바른괄호/solution.py
class Solution(object):
    @staticmethod
    def solution(s):
        test = list(s)
        last_str=test[0]
        before = [last_str]
        for i in range(1,len(test)):
            c = test[i]
            if last_str== '(' and c==')':
                before.pop()
                last_str=  before[-1] if len(before)>0 else ''
            else:
                last_str=c
                before.append(c)
        return len(before) == 0
